File: projects/MusicGame/Music-Game-main/src/MainMenu.java

import javax.swing.*;
import java.awt.event.*;
import java.awt.*;

public class MainMenu extends JFrame implements ActionListener{
    private JPanel scene;
    private Color myColor = new Color(245,245,220);
    private Color buttonColor = new Color(90,114,255);
    @Override
    public void actionPerformed(ActionEvent e){
        SwingUtilities.invokeLater(new MusicGame());
        //Closes main menu window after starting game
        setVisible(false);
    }
    
    public MainMenu(){
        setTitle("LET'S PLAY MUSIC GAME!!");
        setSize(600,400);
        setDefaultCloseOperation(EXIT_ON_CLOSE);
        //JPanel set up
        scene = new JPanel();
        add(scene);
        scene.setBackground(myColor);
        //Strings to print and font
        JLabel title = new JLabel("     MUSIC GAME");
        JLabel buttonspace = new JLabel("                           ");
        JLabel instructions = new JLabel("     INSTRUCTIONS:     ");
        JLabel inst1 =  new JLabel("1.) Press the Play Button                                    ");
        JLabel inst2 =  new JLabel("2.) Hit the Letf and Right Arrow Keys to move");
        JLabel inst3 =  new JLabel("3.) Land on Platforms to Play Notes                  ");
        JLabel inst4 =  new JLabel("4.) Run into Power Ups for a New Experience  ");
        JLabel inst5 =  new JLabel("5.) DON'T FALL DOWN!!!!!                              ");
        Font newfont = new Font("Serif", Font.BOLD, 48);
        Font otherfont = new Font("Serif",Font.BOLD, 19);
        //Create button
        JButton b = new JButton("LET'S PLAY!!!");
        //Print stuff and button edits 
        title.setFont(newfont);
        title.setForeground(buttonColor);
        scene.add(title);
        scene.add(buttonspace);
        scene.add(b);
        instructions.setFont(newfont);
        instructions.setForeground(buttonColor);
        scene.add(instructions);
        inst1.setFont(otherfont);
        inst1.setForeground(buttonColor);
        scene.add(inst1);
        inst2.setFont(otherfont);
        inst2.setForeground(buttonColor);
        scene.add(inst2);
        inst3.setFont(otherfont);
        inst3.setForeground(buttonColor);
        scene.add(inst3);
        inst4.setFont(otherfont);
        inst4.setForeground(buttonColor);
        scene.add(inst4);
        inst5.setFont(otherfont);
        inst5.setForeground(buttonColor);
        scene.add(inst5);
        b.setBackground(buttonColor);
        b.addActionListener(this);
        
        //Centers main menu window
        setLocationRelativeTo(null);
        
        setVisible(true);
    }
    
}

File: projects/MusicGame/Music-Game-main/src/MainMenu.js
class MainMenu {
    constructor() {
        this.myColor = 'rgb(245, 245, 220)';
        this.buttonColor = 'rgb(90, 114, 255)';
        this.init();
    }

    init() {
        document.title = "LET'S PLAY MUSIC GAME!!";
        document.body.style.backgroundColor = this.myColor;
        document.body.style.textAlign = 'center';
        document.body.style.fontFamily = 'Serif';

        const title = this.createLabel("MUSIC GAME", 48, this.buttonColor);
        const buttonspace = this.createLabel(" ", 48, this.buttonColor);
        const instructions = this.createLabel("INSTRUCTIONS:", 48, this.buttonColor);
        const inst1 = this.createLabel("1.) Press the Play Button", 19, this.buttonColor);
        const inst2 = this.createLabel("2.) Hit the Left and Right Arrow Keys to move", 19, this.buttonColor);
        const inst3 = this.createLabel("3.) Land on Platforms to Play Notes", 19, this.buttonColor);
        const inst4 = this.createLabel("4.) Run into Power Ups for a New Experience", 19, this.buttonColor);
        const inst5 = this.createLabel("5.) DON'T FALL DOWN!!!!!", 19, this.buttonColor);

        const playButton = document.createElement('button');
        playButton.innerText = "LET'S PLAY!!!";
        playButton.style.backgroundColor = this.buttonColor;
        playButton.style.color = 'white';
        playButton.style.fontSize = '24px';
        playButton.style.padding = '10px 20px';
        playButton.style.border = 'none';
        playButton.style.cursor = 'pointer';
        playButton.addEventListener('click', this.actionPerformed.bind(this));

        document.body.appendChild(title);
        document.body.appendChild(buttonspace);
        document.body.appendChild(playButton);
        document.body.appendChild(instructions);
        document.body.appendChild(inst1);
        document.body.appendChild(inst2);
        document.body.appendChild(inst3);
        document.body.appendChild(inst4);
        document.body.appendChild(inst5);
    }

    createLabel(text, fontSize, color) {
        const label = document.createElement('div');
        label.innerText = text;
        label.style.fontSize = `${fontSize}px`;
        label.style.color = color;
        label.style.margin = '10px 0';
        return label;
    }

    actionPerformed() {
        // Start the MusicGame
        new MusicGame();
        // Hide the main menu
        document.body.style.display = 'none';
    }
}

// Assuming MusicGame is defined elsewhere
class MusicGame {
    constructor() {
        console.log("Music Game Started!");
    }
}

// Initialize the main menu
new MainMenu();